docs/logging_snippets.py

```diff
@@ -0,0 +1,369 @@
+need to be deleted during teardown.
+"""
+
+import time
+
+from google.cloud.logging.client import Client
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+def do_something_with(item):  # pylint: disable=unused-argument
+    pass
+
+
+# pylint: disable=reimported,unused-variable,unused-argument
+@snippet
+def instantiate_client(_unused_client, _unused_to_delete):
+    """Instantiate client."""
+
+    # [START client_create_default]
+    from google.cloud import logging
+    client = logging.Client()
+    # [END client_create_default]
+
+    credentials = object()
+    # [START client_create_explicit]
+    from google.cloud import logging
+    client = logging.Client(project='my-project', credentials=credentials)
+    # [END client_create_explicit]
+# pylint: enable=reimported,unused-variable,unused-argument
+
+
+@snippet
+def client_list_entries(client, to_delete):  # pylint: disable=unused-argument
+    """List entries via client."""
+
+    # [START client_list_entries_default]
+    for entry in client.list_entries():  # API call(s)
+        do_something_with(entry)
+    # [END client_list_entries_default]
+
+    # [START client_list_entries_filter]
+    FILTER = "logName:log_name AND textPayload:simple"
+    for entry in client.list_entries(filter_=FILTER):  # API call(s)
+        do_something_with(entry)
+    # [END client_list_entries_filter]
+
+    # [START client_list_entries_order_by]
+    from google.cloud.logging import DESCENDING
+    for entry in client.list_entries(order_by=DESCENDING):  # API call(s)
+        do_something_with(entry)
+    # [END client_list_entries_order_by]
+
+    # [START client_list_entries_paged]
+    iterator = client.list_entries()
+    pages = iterator.pages
+
+    page1 = next(pages)  # API call
+    for entry in page1:
+        do_something_with(entry)
+
+    page2 = next(pages)  # API call
+    for entry in page2:
+        do_something_with(entry)
+    # [END client_list_entries_paged]
+
+
+# @snippet  Commented because we need real project IDs to test
+def client_list_entries_multi_project(
+        client, to_delete):  # pylint: disable=unused-argument
+    """List entries via client across multiple projects."""
+
+    # [START client_list_entries_multi_project]
+    PROJECT_IDS = ['one-project', 'another-project']
+    for entry in client.list_entries(project_ids=PROJECT_IDS):  # API call(s)
+        do_something_with(entry)
+    # [END client_list_entries_multi_project]
+
+
+@snippet
+def logger_usage(client, to_delete):
+    """Logger usage."""
+    LOG_NAME = 'logger_usage_%d' % (_millis())
+
+    # [START logger_create]
+    logger = client.logger(LOG_NAME)
+    # [END logger_create]
+    to_delete.append(logger)
+
+    # [START logger_log_text]
+    logger.log_text("A simple entry")  # API call
```

Comment:
PEP 8 [says about string quotes](https://www.python.org/dev/peps/pep-0008/#string-quotes): "In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it."   

"Double quotes for text intended for humans, single quotes for text only used in code" is a rule, and one we have been using (or so I thought).